src/h3lib/lib/polygon.c

```diff
@@ -93,3 +95,172 @@ bool pointInsidePolygon(const GeoPolygon* geoPolygon, const BBox* bboxes,
 #undef IS_EMPTY
 #undef INIT_ITERATION
 #undef ITERATE
+
+/**
+ * Count the number of polygons containing a given loop.
+ * TODO: It would be really nice to memoize this.
+ * @param  loop         Loop to count containers for
+ * @param  polygons     Polygons to test
+ * @param  polygonCount Number of polygons in the test array
+ * @return              Number of polygons containing the loop
+ */
+static int countContainers(LinkedGeoLoop* loop, LinkedGeoPolygon** polygons,
+                           BBox** bboxes, int polygonCount) {
+    int containerCount = 0;
+    for (int i = 0; i < polygonCount; i++) {
+        if (pointInsideLinkedGeoLoop(polygons[i]->first, bboxes[i],
+                                     &loop->first->vertex)) {
+            containerCount++;
+        }
+    }
+    return containerCount;
+}
+
+/**
+ * Find the polygon to which a given hole should be allocated. Note that this
+ * function will return null if no parent is found.
+ * @param  loop         Inner loop describing a hole
+ * @param  polygon      Head of a linked list of polygons to check
+ * @param  polygonCount Number of polygons to check
+ * @return              Pointer to parent polygon, or null if not found
+ */
+static LinkedGeoPolygon* findPolygonForHole(LinkedGeoLoop* loop,
+                                            LinkedGeoPolygon* polygon,
+                                            BBox* bboxes, int polygonCount) {
+    // Early exit with no polygons
+    if (polygonCount == 0) {
+        return NULL;
+    }
+    // Initialize arrays for candidate loops and their bounding boxes
+    LinkedGeoPolygon** candidates =
```

Comment:
`calloc` might be overkill if you know that you will overwrite the whole array with new values.

Reply:
If I don't use `calloc` I can't refer to `candidates[0]` without guards... I'll see if this can be changed to `malloc`.

Reply:
Switch memory assignments to `malloc`.

Reply:
Nice. Just trying to reduce the `calloc` throughout. I have heard using `calloc` is actually a bad idea when using a tool like Valgrind or AddressSanitizer. The zeroing suppresses errors that are likely due to uninitialized data. Like this:

```
int* array = calloc(10, sizeof(int));
// TODO: Load values of array from file.
for (int i = 0; i < 10; i++) {
    array[i] *= 5;
}
```

In the above case, the initialization is marked as `TODO`, but Valgrind won't warn about it because it sees you set the values to zero. On the other hand, using `malloc` would have caused Valgrind to warn about an invalid access to uninitialized data and reminded the programmer to implement the `TODO` behavior.